compiler/native/compile.go

```diff
@@ -5,14 +5,86 @@
 package native
 
 import (
+	"bytes"
+	"encoding/json"
 	"fmt"
+	"io/ioutil"
+	"net/http"
 	"strings"
 
+	"github.com/go-vela/types/library"
 	"github.com/go-vela/types/pipeline"
+	"github.com/go-vela/types/yaml"
 )
 
+// ModifyRequest contains the payload passed to the modification endpoint
+type ModifyRequest struct {
+	Pipeline *yaml.Build `json:"pipeline,omitempty"`
+	Build    int         `json:"build,omitempty"`
+	Repo     string      `json:"repo,omitempty"`
+	User     string      `json:"user,omitempty"`
+}
+
+// modifyConfig sends the configuration to external http endpoint for modification.
+func (c *client) modifyConfig(build *yaml.Build, libraryBuild *library.Build, repo *library.Repo) (*yaml.Build, error) {
+	// create request to send to endpoint
+	modReq := &ModifyRequest{
+		Pipeline: build,
+		Build:    libraryBuild.GetNumber(),
+		Repo:     repo.GetName(),
+		User:     libraryBuild.GetAuthor(),
+	}
+
+	// marshal json to send in request
+	b, err := json.Marshal(modReq)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal modify payload")
+	}
+
+	// create POST request
+	req, err := http.NewRequest("POST", c.ModificationService.Endpoint, bytes.NewBuffer(b))
+	if err != nil {
+		return nil, err
+	}
+
+	// create the client with timeouts
+	client := &http.Client{
+		Timeout: c.ModificationService.Timeout,
+	}
+
+	// add content-type and auth headers
+	req.Header.Add("Content-Type", "application/json")
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", c.ModificationService.Secret))
+
+	// send the request
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	// fail if the response code was not 200
+	if resp.StatusCode != http.StatusOK {
```

Comment:
Are there any other status codes we should be checking for? 

I'm wondering if there would ever be a world where modification failed but the compile process should continue.

Reply:
I also contemplated failure scenarios and potentially processing with the original configuration, but opted with this approach since my use case would be invalidated if the modification to the configuration was not successful. Suppose we could consider adding some configuration within the server to allow them to define some array of status codes to allow. Thoughts?

Reply:
Makes sense. I think if/when we add more codes I lean towards just standardizing on them and not making them configurable but I could be convinced otherwise.

My perspective on not allowing codes to be configured and having a standard is; it's one less thing the admin needs to worry about when designing their modification service. Which I think would reduce the entry point into writing them. I also think it would be better because then integrations could be more shareable if administrators want to open source their validation engines.